core/Cargo.toml

```diff
@@ -172,6 +172,7 @@ bytes = "1.2"
 chrono = "0.4.24"
 dashmap = { version = "5.4", optional = true }
 dirs = { version = "5.0.1", optional = true }
+enum-iterator = "1.4.1"
```

Comment:
Maybe we can remove this dep since we don't need `Sequence` anymore.

Reply:
No. It's still requried from thet test coverage for aligning ErrorCode in https://github.com/apache/incubator-opendal/pull/2276/commits/af5e1d0fce614feb7f3113023bfa1c59ce98f10c.

Reply:
I prefer remove this test first, it doesn't worth that.

Reply:
OK. Then We're at the risk of ErrorCode mismatch, or we can select a default error code for it.

Reply:
> or we can select a default error code for it.

We can return `Unexpected` error for this case. It's easy to be catched and fixed which means we don't need to do defensive programming here.